lib/pipeline/shared.go

```diff
@@ -215,10 +216,13 @@ func generateContainersAndVolumes(obj *unstructured.Unstructured, workflowType v
 
 // TODO(breaking) change this to {promiseIdentifier}-{pipelineType}-pipeline-{short-uuid}
 // for consistency with other resource names (e.g. service account)
-func pipelineName(pipelineType, promiseIdentifier string) string {
-	return pipelineType + "-pipeline-" + promiseIdentifier + "-" + getShortUuid()
-}
+func pipelineName(promiseIdentifier, resourceIdentifier, name, pipelineName string) string {
```

Comment:
is the mapping here:

* promiseIdentifier → promiseName
* resourceIdentifier → resourceName
* name → ??
* pipelineName → pipeliname?

Reply:
This one caught me out in the `PipelineArgs` so could benefit from clarity 

- promiseIdentifier → promiseName ✅
- resourceIdentifier → resourceName ❌
  - The resourceIdentifier, which we use as the `resource-request-id` features both the promise name and resource name e.g. `my-promise-example-request`.
- The name, however, is the name of the object and could thus be `resourceName` _or_ `PromiseName`. I think `objectName` could shed some light on this reduce some abiguity?